sos/plugins/openstack_instack.py

```diff
@@ -28,6 +30,16 @@ def setup(self):
         self.add_copy_spec("/home/stack/instackenv.json")
         self.add_copy_spec("/home/stack/undercloud.conf")
 
+        # collect undercloud-debug: undercloud-debug.log will be created so we
+        # need to generate it in a tmp.dir, collect it now and then cleanup it
+        tmpdir = gettempdir()
+        self.get_cmd_output_now("su -s /bin/sh -c 'source ~/stackrc && " +
+                                "/usr/libexec/openstack-tripleo/" +
+                                "undercloud-debug' stack",
+                                timeout=600, runat=tmpdir,
+                                suggest_filename="undercloud-debug.log")
+        remove("%s/undercloud-debug.log" % tmpdir)
```

Comment:
The only tripleo specific output is collected by the heat loop looking for failed deployments in the overcloud :
https://github.com/openstack/tripleo-incubator/blob/master/scripts/undercloud-debug#L46-L53

Everything else should be collected in the service specific sos plugins. For example, the following ironic node and port show commands should be in the openstack_ironic plugin :
https://github.com/openstack/tripleo-incubator/blob/master/scripts/undercloud-debug#L39-L40

The nova plugin has a good example of how to do this here : 
https://github.com/sosreport/sos/blob/master/sos/plugins/openstack_nova.py#L34

I'm pretty snowed under at the moment but I might be able to land some of these later in the week if we don't have any other volunteers.